wordpress-mobile/WordPress-Android
# Add 10mb limit to loading thumbnails from videos

Fixes one possible reason of #9997

I've found out one of the reasons for the OOMs. The reason is that in order to show the video thumbnails in all the media screens we use glide. Glide starts loading the whole video and allocates memory in order to keep it. If the video is bigger than available memory, we get OOM. This crash happens right after you enter the screen (before the whole video is loaded). In order to fix this I've added a check that pings the URL to find out the size of the video before it's checked. We don't show thumbnails for videos bigger than 10 megabytes. It's still possible to click on the video to see the thumbnail. I think this approach is a good compromise. I don't think users would expect all the videos on the screen to be loaded.

To test (media screen):
- Go to a site with some big videos uploaded
- Go to My Site/Media
- Notice the app hasn't crashed
- Notice the big videos don't show a thumbnail, instead you can see a grey background (in the light mode)


To test (GB post):
- Go to a site with some big videos uploaded
- Click on "Add post"
- Add a "Video" block
- Click on "Add video"
- Choose from Wordpress library
- Notice no crash
- Notice thumbnails get loaded only for videos < 10 mbs


To test (Story):
- Go to a site with some big videos uploaded
- Click on "Add a story"
- Click on "Wordpress" icon
- Notice no crash when showing the list of items
- Notice thumbnails get loaded only for videos < 10 mbs

PR submission checklist:
